### PR TITLE
Deprecating `Functions.singletonList`

### PR DESCRIPTION
**Pull request by jglick**
See https://github.com/jenkinsci/jenkins/pull/6732#discussion_r912016395.

### Proposed changelog entries

* Deprecating `Functions.singletonList` which is not normally needed in JEXL.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6740"></a>

### CODE REVIEW

File: core/src/main/java/hudson/Functions.java
Comment (NotMyFault) [inline core/src/main/java/hudson/Functions.java:1945]:
```suggestion
    @Deprecated(since = "TODO")
```

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/java/hudson/Functions.java
Comment (jglick) [reply core/src/main/java/hudson/Functions.java:1945]: Not sure why it would matter. It could be considered retroactively deprecated since the beginning of the Hudson project. Anyway the call sites are in Jelly sources where no IDE will be able to display this information.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by MarkEWaite** [comment]
I believe that the emailext-template-plugin groovy UI script that calls `h.singletonList(it)` needs to be changed in order to fix [JENKINS-69153](https://issues.jenkins.io/browse/JENKINS-69153).  [Pull request 71](https://github.com/jenkinsci/emailext-template-plugin/pull/71) has been submitted to the plugin with a proposed fix.

**Pull request merged**
